api/requestParams.d.ts

```diff
@@ -0,0 +1,1895 @@
+export interface Generic {
+  method?: string;
+  ignore?: number | number[];
+  filter_path?: string | string[];
+  pretty?: boolean;
+  human?: boolean;
+  error_trace?: boolean;
+  source?: string;
+}
+
+export interface Bulk extends Generic {
+  index?: string;
+  type?: string;
+  wait_for_active_shards?: string;
+  refresh?: 'true' | 'false' | 'wait_for';
+  routing?: string;
+  timeout?: string;
+  fields?: string | string[];
+  _source?: string | string[];
+  _source_exclude?: string | string[];
+  _source_include?: string | string[];
+  pipeline?: string;
+  body: any;
+}
+
+export interface CatAliases extends Generic {
+  name?: string | string[];
+  format?: string;
+  local?: boolean;
+  master_timeout?: string;
+  h?: string | string[];
+  help?: boolean;
+  s?: string | string[];
+  v?: boolean;
+}
+
+export interface CatAllocation extends Generic {
+  node_id?: string | string[];
+  format?: string;
+  bytes?: 'b' | 'k' | 'kb' | 'm' | 'mb' | 'g' | 'gb' | 't' | 'tb' | 'p' | 'pb';
```

Comment:
Create type `Bytes` enum and reference that? as well as for the other inline unions.

Reply:
I had a chat with @timroes, and we agreed that since there is not a real benefit on generalizing this *(the autocompletion would still work)* and the code generation complexity will increase a lot, for now, I'll leave this as it is :)